### PR TITLE
use dispatchRequestEx for requestFollowConversation

### PR DESCRIPTION
**Pull request by flootr**
#### Changes proposed in this Pull Request

* Use `dispatchRequestEx` for `requestConversationFollow`

#### Testing instructions

* In Reader open a post with comments
* Hit _Follow Conversation_  - Does it work? Is the state persisted even after reload?
* Repeat, but deactivate your Wifi in devtools
* Does it show a proper error notice?
* Go online again and retry, does it work?

related #25121

### CODE REVIEW

File: client/state/data-layer/wpcom/read/sites/posts/follow/index.js [hunk 1]
```diff
@@ -13,15 +13,15 @@ import { translate } from 'i18n-calypso';
...
 import { bypassDataLayer } from 'state/data-layer/utils';
 import getReaderConversationFollowStatus from 'state/selectors/get-reader-conversation-follow-status';
 
 import { registerHandlers } from 'state/data-layer/handler-registry';
 
-export function requestConversationFollow( { dispatch, getState }, action ) {
+export const requestConversationFollow = action => ( dispatch, getState ) => {
 	const actionWithRevert = merge( {}, action, {
 		meta: {
 			previousState: getReaderConversationFollowStatus( getState(), {
```
Comment (jsnajdr) [inline client/state/data-layer/wpcom/read/sites/posts/follow/index.js]: Similar opportunity as in https://github.com/Automattic/wp-calypso/pull/28006/files#r227279120 -- set `meta.previousState` in action creator.

File: client/state/data-layer/wpcom/read/sites/posts/follow/index.js [hunk 2]
```diff
@@ -41,50 +41,44 @@ export function requestConversationFollow( { dispatch, getState }, action ) {
...
+	} );
+};
 
-export function receiveConversationFollowError(
-	{ dispatch },
-	{ payload: { siteId, postId }, meta: { previousState } }
-) {
-	dispatch(
+export function receiveConversationFollowError( {
+	payload: { siteId, postId },
```
Comment (jsnajdr) [inline client/state/data-layer/wpcom/read/sites/posts/follow/index.js]: The concept of `action.payload` is not used much in the rest of Calypso -- we just set properties directly on the action object these days. It comes from [Flux Standard Actions](https://github.com/redux-utilities/flux-standard-action) spec.

Comment (flootr) [reply client/state/data-layer/wpcom/read/sites/posts/follow/index.js]: Let's do it in another PR 👍 it affects more files than related to this PR.

**Review by jsnajdr — APPROVED** [review]
Very nice! I posted one nit about `bypassDataLayer`. Nothing is broken and there is no observable behavior change, just a redundant function call 🙂 

Thanks for working on this migration project :+1:

**Pull request merged**